mlrun/utils/notifications/notification_pusher.py

```diff
@@ -36,55 +46,148 @@ def __init__(self, runs: typing.Union[mlrun.lists.RunList, list]):
+                when_state == run_state == "completed"
                 and (not condition or ast.literal_eval(condition))
-            ) or (when_state == "failure" and run_state == "error"):
+            ) or when_state == run_state == "error":
                 return True
 
         return False
 
-    def _load_notification(self, notification_config: dict) -> NotificationBase:
-        params = notification_config.get("params", {})
+    def _load_notification(
+        self, run: mlrun.model.RunObject, notification: mlrun.model.Notification
+    ) -> NotificationBase:
+        name = notification.name
         notification_type = NotificationTypes(
-            notification_config.get("type", "console")
+            notification.kind or NotificationTypes.console
         )
-        if notification_type not in self._notifications:
+        notification_key = f"{run.metadata.uid}-{name or notification_type}"
+        if notification_key not in self._notifications:
             self._notifications[
-                notification_type
-            ] = notification_type.get_notification()(params)
+                notification_key
+            ] = notification_type.get_notification()(name, notification.params)
         else:
-            self._notifications[notification_type].load_notification(params)
+            self._notifications[notification_key].load_notification(notification.params)
+
+        logger.debug(
+            "Loaded notification", notification=self._notifications[notification_key]
+        )
+        return self._notifications[notification_key]
+
+    async def _send_notification(
+        self,
+        notification: NotificationBase,
+        run: mlrun.model.RunObject,
+        notification_object: mlrun.model.Notification,
+        db: mlrun.api.db.base.DBInterface,
+    ):
+        message = self.messages.get(run.state(), "")
+        severity = (
+            notification_object.severity or mlrun.api.schemas.NotificationSeverity.INFO
+        )
+        logger.debug(
+            "Sending notification",
+            notification=notification_object.to_dict(),
+            run_uid=run.metadata.uid,
+        )
+        try:
+            if asyncio.iscoroutinefunction(notification.send):
+                await notification.send(message, severity, [run.to_dict()])
+            else:
+                notification.send(message, severity, [run.to_dict()])
 
-        return self._notifications[notification_type]
+            if mlrun.config.is_running_as_api():
+                await self._update_notification_status(
+                    db,
+                    run.metadata.uid,
+                    run.metadata.project,
+                    notification_object,
+                    status=mlrun.api.schemas.NotificationStatus.SENT,
+                    sent_time=datetime.datetime.now(tz=datetime.timezone.utc),
+                )
+        except Exception as exc:
+            if mlrun.config.is_running_as_api():
```

Comment:
if not running in API, why we wouldn't want to update the notification as well?

Reply:
If not running in API the whole notification mechanism is different. They are fired locally (from jupyter) - no notification in the DB for the monitor to find, no secrets as it's all local.